src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -950,13 +950,16 @@ private async ValueTask RefreshInstalledAndUpdatesTabsAsync()
 
         private async Task<(int, int)> GetInstalledVulnerableAndDeprecatedPackagesCountAsync(PackageLoadContext loadContext, CancellationToken token)
         {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
The call here arrives on the UI thread, maybe we just capture it there and pass it to this method? 

It's largely stylistic, as while adding extra await isn't great, the cost is so minimal that it doesn't matter.

Reply:
I got this suggestion from @davkean to make sure we're on the UI thread here. You're right that it comes in on the UI thread in this particular case, but there's nothing enforcing that to be true, so we use the SwitchToMainThreadAsync to ensure that no matter what the caller thread is, we are on the UI thread when accessing the UI property. Calling SwitchToMainThreadAsync is a noop in this case and the cost is minimal.

Reply:
I think you misunderstood my suggestion.

I'm not saying just assume you're on the UI thread, but rather pass the source as a parameter to the method.

Reply:
I see what you mean now! Yeah, that makes sense, especially since this method is private and has only one reference, it won't be affecting anything to change its signature.